// meewmeewapi/lib/Aio.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const Base_1 = __importDefault(require("./Base"));
class Aio extends Base_1.default {
    constructor(apikey) {
        super(apikey);
    }
    /**
     *
     * @param videoURL All url of video
     * @returns Promise<any>
     */
    listURL(videoURL) {
        var _ = this;
        return new Promise(function (resolve, reject) {
            _.axios.get(`${_.ApiURLv2}/aio`, {
                params: {
                    apikey: _.apikey,
                    version: _.version,
                    url: encodeURIComponent(videoURL),
                }
            }).then(function ({ data }) {
                return resolve(data);
            }).catch(function (error) {
                return reject(error);
            });
        });
    }
    /**
     *
     * @param downloadURL DownloadURL in list of response from listURL
     * @param pathFile Path to save
     * @returns Promise<any>
     */
    download(downloadURL, pathFile) {
        var _ = this;
        return new Promise(function (resolve, reject) {
            if (pathFile)
                return _.writeStream(downloadURL, pathFile, resolve, reject);
            else
                return resolve(downloadURL);
        });
    }
}
exports.default = Aio;
