// meewmeewapi/lib/Image.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const Base_1 = __importDefault(require("./Base"));
class Image extends Base_1.default {
    constructor(apikey) {
        super(apikey);
    }
    /**
     *
     * @param imageType 'meow' | 'dog' | 'sexy' | 'girl' | 'boy' | 'wibu' | 'cosplay'
     * @param path string
     * @returns Promise<ArrayBuffer>
     */
    random(imageType, path) {
        let { writeFile, apikey, axios, ApiURLv2, version } = this;
        return new Promise(function (resolve, reject) {
            axios.get(`${ApiURLv2}/image`, {
                params: {
                    type: imageType,
                    apikey: apikey,
                    version: version
                }
            }).then(function ({ data }) {
                if (path && data.success)
                    return writeFile(data.data, path, resolve, reject);
                return resolve(data);
            }).catch(function (error) {
                return reject(error);
            });
        });
    }
}
exports.default = Image;
