// meewmeewapi/lib/Simsimi.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const Base_1 = __importDefault(require("./Base"));
class Simsimi extends Base_1.default {
    constructor(apikey) {
        super(apikey);
    }
    /**
     *
     * @param text string
     * @returns Object<{ success: boolean, msg: string }>
     */
    chat(text) {
        var _ = this;
        return new Promise(function (resolve, reject) {
            _.axios.get(`${_.ApiURLv2}/simsimi/chat`, {
                params: {
                    ask: encodeURIComponent(text),
                    apikey: _.apikey,
                    version: _.version
                }
            }).then(function ({ data }) {
                return resolve(data);
            }).catch(function (error) {
                return reject(error);
            });
        });
    }
    /**
     *
     * @param ask string
     * @param answer string
     * @returns Object<{ success: boolean, msg: string }>
     */
    teach(ask, answer) {
        var _ = this;
        return new Promise(function (resolve, reject) {
            _.axios.post(`${_.ApiURLv2}/simsimi/teach`, null, {
                params: {
                    ask: encodeURIComponent(ask),
                    answer: encodeURIComponent(answer),
                    apikey: _.apikey,
                    version: _.version
                }
            }).then(function ({ data }) {
                return resolve(data);
            }).catch(function (error) {
                return reject(error);
            });
        });
    }
}
exports.default = Simsimi;
