// meewmeewapi/lib/Weather.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const Base_1 = __importDefault(require("./Base"));
class Weather extends Base_1.default {
    constructor(apikey) {
        super(apikey);
    }
    /**
     *
     * @param location Location
     * @param path path to save
     * @returns Promise<any>
     */
    image(location, path) {
        let { writeStream, axios, ApiURLv2, apikey, version } = this;
        return new Promise(function (resolve, reject) {
            axios.get(`${ApiURLv2}/weather`, {
                params: {
                    location: encodeURI(location),
                    apikey: apikey,
                    version: version
                }
            }).then(function ({ data }) {
                if (path && data.success)
                    return writeStream(data.data, path, resolve, reject);
                return resolve(!data.success ? data : {
                    success: true,
                    data: data.data
                });
            }).catch(function (error) {
                return reject(error);
            });
        });
    }
}
exports.default = Weather;
