// meewmeewapi/lib/Word.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const Base_1 = __importDefault(require("./Base"));
class Word extends Base_1.default {
    constructor(apikey) {
        super(apikey);
    }
    /**
     *
     * @param text string
     * @param lang vi | en
     * @returns Object<{success: boolean, data: string}>
     */
    linkWord(text, lang = 'vi') {
        var _ = this;
        return new Promise(function (resolve, reject) {
            _.axios.get(`${_.ApiURLv2}/word/linkword`, {
                params: {
                    ask: text,
                    lang: lang,
                    apikey: _.apikey,
                    version: _.version
                }
            }).then(function ({ data }) {
                return resolve(data);
            }).catch(function (error) {
                return reject(error);
            });
        });
    }
    /**
     *
     * @param level string
     * @returns Object<{correct: string, random: Array<string>}>
     */
    sortWord(level = 'random') {
        var _ = this;
        return new Promise(function (resolve, reject) {
            _.axios.get(`${_.ApiURLv2}/word/random`, {
                params: {
                    level: level,
                    apikey: _.apikey,
                    version: _.version
                }
            }).then(function ({ data }) {
                return resolve(data);
            }).catch(function (error) {
                return reject(error);
            });
        });
    }
}
exports.default = Word;
